src/Evolution/Initialization/Initialize.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Parallel/Actions/TerminatePhase.hpp"
+#include "Parallel/AddOptionsToDataBox.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Initialization {
+/// Available actions to be used in the initialization.
+namespace Actions {
+template <typename OptionsTagsToRemove>
+struct RemoveOptionsFromDataBox {
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ActionList, typename ArrayIndex,
+            typename ParallelComponent>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using options_tags_to_remove_this_time = tmpl::filter<
+        OptionsTagsToRemove,
+        tmpl::bind<tmpl::list_contains, tmpl::pin<DbTagsList>, tmpl::_1>>;
+    return std::make_tuple(
+        db::create_from<options_tags_to_remove_this_time>(std::move(box)));
+  }
+};
+}  // namespace Actions
+
+namespace detail {
+template <typename Action, typename = cpp17::void_t<>>
+struct get_options_tags {
+  using type = tmpl::list<>;
+};
+
+template <typename Action>
+struct get_options_tags<
+    Action, cpp17::void_t<typename Action::initialization_option_tags>> {
+  using type = typename Action::initialization_option_tags;
+};
+
+template <typename ActionsList>
+using options_databox_tags = tmpl::remove_duplicates<tmpl::flatten<
+    tmpl::transform<ActionsList, detail::get_options_tags<tmpl::_1>>>>;
+}  // namespace detail
+
+/// Given the list of initialization actions generates the class that
+/// `add_options_to_databox` should be set to in the parallel component.
+template <typename ActionsList>
+using generate_add_options_to_databox = Parallel::ForwardAllOptionsToDataBox<
+    detail::options_databox_tags<ActionsList>>;
+
+/// Generate the final list of initialization actions from the requested
+/// initialization actions.
+///
+/// This adds actions to remove the options that were added to the DataBox and
+/// terminates the phase. None of the initialization actions should terminate
+/// the algorithm.
+template <typename ActionsList>
+using generate_action_list =
```

Comment:
This might be okay for `generate_add_options_to_databox`, but not for `generate_action_list` since there specific actions are added. Please keep in mind that telling someone to use `generate_action_list<Actions...>` is easier for them than looking at the `tmpl::append` code. This is **user interface** code, and so the goal should be to make it not scary to use and easy to read.

Reply:
I agree we should make the interface nice for the user! Indeed I think it's actually more transparent and understandable for the user to add `Parallel::ForwardAllOptionsToDataBox` somewhere, and then in the initialization action list explicitly add a `RemoveOptionsFromDatabox` and then a `TerminatePhase`. It's quite nice that the PDALs quite explicitly tell the story what the program will do, and I'd like to avoid hiding that in wrappers. Thoughts?

Reply:
I'm worried people will forget about the `RemoveOptionsFromDatabox` and `TerminatePhase`, which will lead to extra things in the DataBox for no reason, and possibly inexplicable infinite loops at runtime. I'd rather prevent those two bugs from appearing at all.

Reply:
I understand your worry. I‘d nevertheless prefer the transparency, since users must know how action lists work for writing an executable.. I‘ll think about it a bit more, we can talk tomorrow

Reply:
We decided in a call to have a `RemoveOptionsAndTerminatePhase` action that takes no template parameters, so removes all options. We can also expose `RemoveOptions<OptionTags...>` on its own. Also expose `collect_option_tags` so that `Forward...<>` can be used explicitly in the metavariables.